kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/AssetMaintainableImpl.java

```diff
@@ -0,0 +1,100 @@
+import org.kuali.rice.krad.service.NoteService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.sys.KFSConstants;
+
+@SuppressWarnings("deprecation")
+public class AssetMaintainableImpl extends org.kuali.kfs.module.cam.document.AssetMaintainableImpl {
+
+    private static final long serialVersionUID = 5710219294064590806L;
+
+    @Override
+    public void processAfterEdit(MaintenanceDocument document, Map<String, String[]> parameters) {
+        List<Note> notes = new ArrayList<Note>();
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(this.getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.CHANGE);
+        document.setNotes(notes);
+
+        super.processAfterEdit(document, parameters);
+    }
+
+    @Override
+    public void setGenerateDefaultValues(String docTypeName) {
+        super.setGenerateDefaultValues(docTypeName);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(this.getBusinessObject().getObjectId());
+
+            if (notes.isEmpty()) {
+                Note newBONote = getNewBoNoteForAdding(KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+                notes.add(newBONote);
+            }
+        }
+    }
+
+    @Override
+    public void setupNewFromExisting(MaintenanceDocument document, Map<String, String[]> parameters) {
+        super.setupNewFromExisting(document, parameters);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(this.getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+
+        document.setNotes(notes);
+    }
+
+    /**
+     * Checks whether the previous note was an "Add" with the same document number as this one
+     *
+     * @param prefix
+     *            String to determine if it is a note "Add" or a note "Change"
+     */
+    private void setAssetCreateAndUpdateNote(List<Note> notes, String prefix) {
+        boolean shouldAddNote = true;
+
+        if (prefix.equals(KFSConstants.CreateAndUpdateNotePrefixes.CHANGE)) {
+            // Check whether the previous note was an "Add" with the same document number as this one
+            if (!notes.isEmpty()) {
+                Note previousNote = notes.get(notes.size() - 1);
+                if (previousNote.getNoteText().contains(getDocumentNumber())) {
+                    shouldAddNote = false;
```

Comment:
After talking with Julie and Scott, it was determined that this code was probably just directly ported from KFS 3. In the sake of time, no need to address this issue if that is how it was functioning in KFS 3. Thanks!